### PR TITLE
Add entries to cloud init

### PR DESCRIPTION
**Pull request by Xenwar**
- Install and setup keepalived
- Pass config file to keepalived
- Install required binaries
- run kubeadm init with VIP
- configure kubectl to access api-server on VIP

### CODE REVIEW

**Comment by nordixinfra** [comment]
Can one of the admins verify this patch?

**Comment by maelk** [comment]
don't push that upstream. If you want, we can create a branch in this repo and you can work and do PR towards that branch for easier collaboration.

**Comment by Xenwar** [comment]
> don't push that upstream. If you want, we can create a branch in this repo and you can work and do PR towards that branch for easier collaboration.

This PR is for use locally within the team and does not have upstream setup.

File: user_data.sh [hunk 1]
```diff
@@ -27,31 +27,101 @@ fi
...
+  - yum install docker-ce docker-ce-cli containerd.io -y
+  - usermod -aG docker metal3
+  - systemctl start docker
+  - systemctl enable docker
+  # Install kubernetes
+  - setenforce 0
+  - sed -i 's/^SELINUX=enforcing$/SELINUX=permissive/' /etc/selinux/config
+  - yum install -y kubelet kubeadm kubectl --disableexcludes=kubernetes
+  - systemctl enable --now kubelet
+  - kubeadm init --apiserver-advertise-address 55.66.77.88 --ignore-preflight-errors=all
```
Comment (Jaakko-Os) [inline user_data.sh]: some kubeadm init options:
--upload-certs could be included here
--certificate-key to pass a self generated key

kubeadm join to be included here with a self-generated token. I can provide the details.

**Comment by Xenwar** [comment]
@maelk Do we still need this issue ?

**Comment by maelk** [comment]
this was merged upstream

**Comment by Xenwar** [comment]
> this was merged upstream

How about this one ?
https://github.com/Nordix/cluster-api-provider-baremetal/pull/12

**Comment by Xenwar** [comment]
Apparently, the content of https://github.com/Nordix/cluster-api-provider-baremetal/pull/12 is also merged.
Quite surprised that another PRs created while my PR is hanging, even with an lgtm.

Not sure why or who created those PRs. We should probably add more audience to this conversation. Will try to setup a meeting with the community on this.

**Comment by maelk** [comment]
This PR is in Nordix. as such it should not be merged, since we follow the upstream first policy. If you recall this PR contains the PoC you did for Keepalived setup. Based on that someone implemented the actual solution. I don't recall exactly who did the actual implementation. From the comments, it seems Jan was the one who did, I'm not sure about that.

**Comment by Xenwar** [comment]
> This PR is in Nordix. as such it should not be merged, since we follow the upstream first policy. If you recall this PR contains the PoC you did for Keepalived setup. Based on that someone implemented the actual solution. I don't recall exactly who did the actual implementation. From the comments, it seems Jan was the one who did, I'm not sure about that.

Please check the content again. This is beyond the keepalived poc, as is indicated even in the branch naming. And, please refer to your own comment as to why I did not create a PR in the upstream. And, will check the commits to get the point at which it was merged.

**Comment by Xenwar** [comment]
@jan-est ?

**Comment by maelk** [comment]
Well if there is something missing please update this branch and we can consider creating a PR upstream

**Comment by Xenwar** [comment]
> Well if there is something missing please update this branch and we can consider creating a PR upstream

Well, that really is not the issue. Thanks again for your reply.

**Comment by maelk** [comment]
Would you mind then explaining clearly what the issue with this PoC branch is ? As far as I can see, the PoC was successfully completed and then implemented upstream, so I have issues understanding what the problem is, sorry for that.

**Comment by Xenwar** [comment]
No worries. We can ignore this for now and focus current tasks. Thanks.

**Pull request closed without merging**